### PR TITLE
UF-7191: Handle if externalMessages are null

### PR DESCRIPTION
**Pull request by cheezi0747**

### CODE REVIEW

File: src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java [hunk 1]
```diff
@@ -1,11 +1,15 @@
 package se.sundsvall.webmessagecollector.integration.opene;
 
+import static java.util.Collections.*;
```
[the remark this answers is not in the document]
Comment (cheezi0747) [reply src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java]: Yeah. IDK what IntelliJ is smoking. Fixed.

File: src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java
Comment (josc04) [inline src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java:1]: Some test to verify handling when externalMessages equals null?

File: src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java [hunk 2]
```diff
@@ -20,51 +23,55 @@
...
+		.appendPattern(" HH:mm")
+		.optionalEnd()
+		.parseDefaulting(ChronoField.HOUR_OF_DAY, 0)
+		.parseDefaulting(ChronoField.MINUTE_OF_HOUR, 0)
+		.toFormatter();
+
+	List<MessageEntity> mapMessages(final byte[] errands, final String familyId) {
+
+		if (errands == null) {
+			return List.of();
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java:41]: You have added a statical import to: java.util.Collections.emptyList
Maybe change to `emptyList()` here?

(`emptyList()` is more intuitive than `List.of()` for this purpose i.m.h.o.)

File: src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java [hunk 3]
```diff
@@ -20,51 +23,55 @@
...
+								.withUsername(externalMessage.getPoster().getUsername())
+								.withUserId(String.valueOf(externalMessage.getPoster().getUserID()));
+						}
+						return entity.build();
+					})
+
+					.toList();
+		} catch (final JsonProcessingException e) {
+			LOG.info("Something went wrong parsing messages", e);
+			return List.of();
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEMapper.java:73]: `emptyList()` ?

File: src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEClient.java [hunk 4]
```diff
@@ -17,50 +17,52 @@
...
+			.setDefaultCredentialsProvider(getCredentials())
+			.build();
+	}
+
+	byte[] getMessages(final String familyId, final String fromDate, final String toDate) throws IOException {
+		return getBytes(buildUrl(properties.getMessagesPath(), Map.of("familyid", familyId), fromDate, toDate));
+	}
+
+	private byte[] getBytes(final URI url) throws IOException, OpenEException {
+		return httpClient.execute(new HttpGet(url), responseHandler -> {
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEClient.java:38]: Feign?

Comment (josc04) [reply src/main/java/se/sundsvall/webmessagecollector/integration/opene/OpenEClient.java:38]: I'm in the process of switching to Feign in another task

**Review by foobar-1337 — APPROVED** [review]

**Pull request merged**